spec/lib/api/v3/work_packages/work_package_representer_spec.rb

```diff
@@ -507,6 +505,39 @@
         end
       end
 
+      context 'when the user does not have the permission to remove watchers' do
+        before do
+          role.permissions.delete(:delete_work_package_watchers) and role.save
+        end
+
+        it 'should not have a link to add watcher' do
+          expect(subject).not_to have_json_path('_links/removeWatcher/href')
+        end
+      end
+
+      describe 'watchers link' do
+        context 'when the user is allowed to see watchers' do
+          it_behaves_like 'has an untitled link' do
+            let(:link) { 'watchers' }
+            let(:href) { api_v3_paths.work_package_watchers work_package.id }
+          end
+
+          it 'embeds the watchers as collection' do
+            is_expected.to be_json_eql('Collection'.to_json).at_path('_embedded/watchers/_type')
```

Comment:
At this place I could imagine having a shared example checking the integrity of a collection. I'm not sure if we have something like that already. What do you think about that?